tests/src/JIT/Regression/JitBlue/GitHub_10780/GitHub_10780.cs

```diff
@@ -0,0 +1,52 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+// Repro case for a bug involving hoisting of static field loads out of
+// loops and (illegally) above the corresponding type initializer calls.
+
+using System.Runtime.CompilerServices;
+
+namespace N
+{
+    public struct Pair
+    {
+        public int Left;
+        public int Right;
+
+        public static Pair TenFour = new Pair() { Left = 10, Right = 4 };
```

Comment:
Yeah, the role of `C.Two` in this test is a bit confusing -- I just needed something in the loop that would keep us from hoisting the type init call (which, btw, is the call to init the statics of `Pair`, *not* the statics of `C`), and needed it to not keep registers live across the loop (lest our costing decide not to hoist the reference to `Pair.TenFour`), so writing to a static (which stops us from hoisting the call because it may throw) was what I landed on.  The bug is in reordering the call to `Pair..cctor` vs the load of`Pair.TenFour`, not vs the store of `C.Two`.